src/test/java/uk/gov/pay/adminusers/unit/service/ServiceResourceUpdateTest.java

```diff
@@ -0,0 +1,312 @@
+        assertCyServiceNameJson("new-cy-name", json);
+    }
+
+    @Test
+    public void shouldSuccess_whenReplaceServiceNameAnd_replaceCyName() {
+
+        ServiceEntity thisServiceEntity = ServiceEntityBuilder.aServiceEntity()
+                .withName("old-en-name")
+                .withServiceNameEntity(SupportedLanguage.WELSH, "old-cy-name")
+                .build();
+        String externalId = thisServiceEntity.getExternalId();
+
+        String jsonPayload = fixture("fixtures/resource/service/patch/update-name-and-update-cy-name.json");
+        when(mockedServiceDao.findByExternalId(externalId)).thenReturn(Optional.of(thisServiceEntity));
+        when(mockedServiceDao.merge(thisServiceEntity)).thenReturn(thisServiceEntity);
+
+        Response response = resources.target(format(API_PATH, thisServiceEntity.getExternalId()))
+                .request()
+                .method("PATCH", Entity.json(jsonPayload));
+
+        assertThat(response.getStatus(), is(200));
+
+        String body = response.readEntity(String.class);
+        JsonPath json = JsonPath.from(body);
+
+        assertThat(json.get("name"), is("new-en-name"));
+        assertEnServiceNameJson("new-en-name", json);
+        assertCyServiceNameJson("new-cy-name", json);
+    }
+
+    @Test
+    public void shouldSuccess_whenUpdateOnlyCyName() {
+
+        ServiceEntity thisServiceEntity = ServiceEntityBuilder.aServiceEntity()
+                .withServiceNameEntity(SupportedLanguage.WELSH, "old-cy-name")
+                .build();
+        String externalId = thisServiceEntity.getExternalId();
+
+        String jsonPayload = fixture("fixtures/resource/service/patch/update-cy-name-only.json");
+        when(mockedServiceDao.findByExternalId(externalId)).thenReturn(Optional.of(thisServiceEntity));
+        when(mockedServiceDao.merge(thisServiceEntity)).thenReturn(thisServiceEntity);
+
+        Response response = resources.target(format(API_PATH, thisServiceEntity.getExternalId()))
+                .request()
+                .method("PATCH", Entity.json(jsonPayload));
+
+        assertThat(response.getStatus(), is(200));
+
+        String body = response.readEntity(String.class);
+        JsonPath json = JsonPath.from(body);
+
+        assertThat(json.get("name"), is("System Generated"));
+        assertCyServiceNameJson("new-cy-name", json);
+    }
+
+    @Test
+    public void shouldError404_ifServiceExternalIdDoesNotExist() {
+        String jsonPayload = fixture("fixtures/resource/service/patch/update-cy-name-only.json");
+        String externalId = "externalId";
+        when(mockedServiceDao.findByExternalId(externalId)).thenReturn(Optional.empty());
+
+        Response response = resources.target(format(API_PATH, externalId))
+                .request()
+                .method("PATCH", Entity.json(jsonPayload));
+
+        assertThat(response.getStatus(), is(404));
+    }
+
+    @Test
+    public void shouldError400_ifFieldValueMissing() {
```

Comment:
This is equivalent to `ServiceResourceUpdateTest.shouldError400_ifMandatoryFieldMissing()`

Reply:
I renamed the tests that are asserting for missing fields to be explicit that is mandatory